### PR TITLE
fixes whitelisted items not showing messages when removing chems if in storage

### LINKED ISSUE #1543: Hypo spray doesnt warn when in storage.
[BUG]

**Describe the bug**
The hypo spray doesn't give the needed warning for harmful chemicals when it is stored in a belt or backpack. It just says the chem is added but that chem is in fact deleted.
**To Reproduce**
1. Have hypo spray
2. Have harmful chem
3. Put hypo spray into backpack
4. Try to fill it
5. Be sad

**Expected behavior**
For the warning to show up rather than just deleting the chem

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by zjdtmkhzt**
[minor]
## About the PR 
fixes #1543 
Uses usr in a proc, maybe someone wants to chime in with a better solution.
## Why's this needed? 
fixes #1543

### CODE REVIEW

**Comment by zjdtmkhzt** [comment]
Actually, I should change this around a bit to allow for calls to this to actually give the user, if they want.
The syringe gun attempts this right now but fails if the user is not directly holding the gun, which makes it a bit pointless in this case, but maybe a call to this will want to do it in the future.

File: code/procs/helpers.dm [hunk 1]
```diff
@@ -2442,8 +2442,9 @@ proc/check_whitelist(var/atom/TA, var/list/whitelist, var/mob/user as mob, var/c
 			TA.reagents.del_reagent(reagent_id)
 			found = 1
 	if (found)
+		if(!user) user = usr // using usr here because some of the procs that call this aren't in a position to know the actual user
 		if (user)
-			boutput(user, "[custom_message]") // haine: done -> //TODO: using usr in procs is evil shame on you
+			boutput(user, "[custom_message]")
 		else if (ismob(TA.loc))
 			var/mob/M = TA.loc
 			boutput(M, "[custom_message]")
```
Comment (ZeWaka) [inline code/procs/helpers.dm:2450]: this should be evaluated before usr imo

Comment (zjdtmkhzt) [reply code/procs/helpers.dm]: I changed it around a bit, is this any better?

Comment (zjdtmkhzt) [reply code/procs/helpers.dm]: Actually, user should probably be checked first, if the calling proc goes through the trouble of sending it.

Comment (zjdtmkhzt) [reply code/procs/helpers.dm:2449]: Alright, changed it again, please comment. ^_^

**Pull request merged**